nginx/modsecurity/ngx_http_modsecurity.c

```diff
@@ -1136,3 +1239,36 @@ ngx_http_modsecurity_drop_action(request_rec *r)
     ctx->r->connection->error = 1;
     return 0;
 }
+
+    static ngx_int_t
+ngx_http_variable_get_modsec_latency_detect_mode(ngx_http_request_t* r,
+    ngx_http_variable_value_t* v, uintptr_t data)
+{
+    return NGX_OK;
+}
+
+    static ngx_int_t
+ngx_http_variable_get_modsec_latency_prevent_mode(ngx_http_request_t* r,
+     ngx_http_variable_value_t* v, uintptr_t data)
+{
+    return NGX_OK;
+}
+
+    static ngx_int_t
+ngx_http_modsecurity_set_modsec_latency(ngx_http_request_t* r,
+    ngx_http_variable_value_t* v, ngx_msec_int_t modsec_latency)
+{
+    if (v->data == NULL) {
+        v->data = ngx_pnalloc(r->pool, NGX_TIME_T_LEN + 4);
```

Comment:
Shouldn't we free this memory back to the pool. If yes where are we doing it? If not why?

Reply:
Memory in a nginx pool is freed automatically when the pool is destroyed. http://nginx.org/en/docs/dev/development_guide.html#pool.  Since the pool allocation and deallocation happens in the core module, when the r->pool will be freed by ngx_pfree, this memory will also be freed. Cannot free the memory in the module since the variable needs to persist and hold the value till the access_logs are generated by the core module.